File: js/promptRefiner.js
/**
 * Prompt Refiner Module
 * Transforms casual user input into professional specifications
 * by calling the backend /api/refinePrompt endpoint.
 */
class PromptRefiner {
    constructor(apiClient) {
        this.api = apiClient;
    }

    /**
     * Refine a user prompt into a professional specification.
     * @param {string} userPrompt - Raw input from the user
     * @returns {Promise<{originalPrompt: string, refinedPrompt: string, timestamp: string}>}
     */
    async refine(userPrompt) {
        if (!userPrompt || !userPrompt.trim()) {
            throw new Error('Please enter a description of what you want to build.');
        }

        const trimmed = userPrompt.trim();
        if (trimmed.length < 10) {
            throw new Error('Please provide a more detailed description (at least 10 characters).');
        }

        const data = await this.api.refinePrompt(trimmed);

        if (data.error) {
            throw new Error(data.error);
        }

        return {
            originalPrompt: trimmed,
            refinedPrompt: data.refinedPrompt || data.result || '',
            timestamp: new Date().toISOString()
        };
    }

    /**
     * Format the refined prompt for display in the UI.
     * @param {object} refinementResult
     * @returns {string}
     */
    formatForDisplay(refinementResult) {
        return refinementResult.refinedPrompt;
    }
}

File: backend/prompt_refiner.py
"""
Prompt Refiner
Transforms a casual user description into a professional frontend specification.
"""
from api_client import chat_completion

_SYSTEM_PROMPT = (
    "You are an expert UI/UX designer and senior web developer. "
    "Your task is to take a user's rough idea for a website or web application "
    "and refine it into a detailed, professional specification.\n\n"
    "Analyse the user's input and provide:\n"
    "1. A clear, enhanced description of what they want to build\n"
    "2. Key features and functionality that should be included\n"
    "3. Suggested design elements and visual style\n"
    "4. Technical requirements (responsive, accessibility, etc.)\n"
    "5. User experience considerations\n\n"
    "Format your response as a structured specification that will be used "
    "to guide code generation. Be thorough but concise."
)


def refine_prompt(user_prompt: str) -> str:
    """
    Refine a raw user prompt into a professional specification.

    Args:
        user_prompt: The user's raw description.

    Returns:
        A refined, detailed specification string.
    """
    messages = [
        {"role": "system", "content": _SYSTEM_PROMPT},
        {"role": "user", "content": user_prompt},
    ]
    return chat_completion(messages, temperature=0.5, max_tokens=1500)
